### ISSUE TITLE
When I set a label for my field in my dbt .yml, it doesn't apply when I add the field on the x-axis

### ISSUE THREAD

**Issue by TuringLovesDeathMetal** [lightdash/lightdash#9298]
### Description

Here's what my .yml looks like:

```yaml
      - name: organization_created_ts
        meta:
          dimension:
            label: organization created
            type: timestamp
```

So, in Lightdash, it appears like this: 
<img width="376" alt="image" src="https://github.com/lightdash/lightdash/assets/31848148/2db6a6c3-1b34-45f6-9299-3c7506690273">

And in the results table, it appears like this:

<img width="522" alt="image" src="https://github.com/lightdash/lightdash/assets/31848148/2d3bbbbd-b52d-46bc-a161-b8f4d394b6ef">

But then, in my chart, the label appears as the raw field name (not the label name):
<img width="825" alt="image" src="https://github.com/lightdash/lightdash/assets/31848148/1b004e87-ad98-4b4d-9826-13c1f4c4f993">

Instead of saying `organization created ts`, it should say `organization created week`

**Comment by joaoviana** [comment]
We've introduced a change here: https://github.com/lightdash/lightdash/issues/8094, to simplify the date x-axis to include the `friendlyName` of the dimension + ignoring the date granularity. This was helpful for the `date zoom` functionality (where at one point we couldn't update the date granularity of the x axis label I believe), but now we can. 

With this change, we will bring back the DATE LABEL + GRANULARITY. 

I've just tested date zooming a chart with the new changes and it changed the granularity correctly. 
<img width="866" alt="Screenshot 2024-03-18 at 12 29 06" src="https://github.com/lightdash/lightdash/assets/7611706/e53225dc-49e7-4058-8626-58a74334460e">
<img width="429" alt="Screenshot 2024-03-18 at 12 28 58" src="https://github.com/lightdash/lightdash/assets/7611706/99b0679b-3dad-4d98-8a36-999e2a97a1da">

sounds good @TuringLovesDeathMetal ?
